### DISCUSSION TITLE
serde support for config data structures

### DISCUSSION THREAD

**Discussion by drHuangMHT** [libp2p/rust-libp2p#5566] [Ideas]
Derive `Serialize` and `Deserialize` for `Config` structs and related data structures?   
I do understand that some data structures cannot be serialized or deserialize. But could we make it happen with simple ones first?

**Comment by jxs** [comment]
yeah looks fine to me if do it as for example in `gossipsub` and feature gate it

**Comment by dariusc93** [comment]
What would your use case be for wanting to derive serde `Serialize` and `Deserialize` for the Config in libp2p vs making your own config that that derives those two with your own fields (besides just reducing redundancy)?

**Reply by drHuangMHT** [reply]
No more use case besides reducing redundancy. Not all fields can be serialized so I'm already doing it manually by copy-pasting. It is not very fun because I have to include the documentation as well.
